101-synapse-poc/azuredeploy.json

```diff
@@ -0,0 +1,621 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "metadata": {
+                "description": "Location for your deployment."
+            },
+            "defaultValue": "[resourceGroup().location]"
+        },
+        "companyTla": {
+            "type": "string",
+            "metadata": {
+                "description": "This is a Three Letter Acronym for your company name. 'CON' for Contoso for example."
+            }
+        },
+
+        "userObjectID": {
+            "type": "string",
+            "metadata": {
+                "description": "This is where you give a User permissions to the Data Lake Storage Account. This will give the user Storage Blob Data Contributor to work with the file system."
+            }
+        },
+        "allowAllConnections": {
+            "type": "string",
+            "allowedValues": [
+                "true",
+                "false"
+            ],
+            "defaultValue": "true"
+        },
+        "sparkDeployment": {
+            "type": "string",
+            "defaultValue": "true",
+            "allowedValues": [
+                "true",
+                "false"
+            ],
+            "metadata": {
+                "description": "'True' deploys an Apache Spark pool as well as a SQL pool. 'False' does not deploy an Apache Spark pool."
+            }
+        },
+        "sparkNodeSize": {
+            "type": "string",
+            "metadata": {
+                "description": "This parameter will determine the node size if SparkDeployment is true"
+            },
+            "defaultValue": "Medium",
+            "allowedValues": [
+                "Small",
+                "Medium",
+                "Large"
+            ]
+        },
+        "deploymentType": {
+            "type": "string",
+            "allowedValues": [
+                "devtest",
+                "poc",
+                "prod",
+                "shared"
+            ],
+            "defaultValue": "poc",
+            "metadata": {
+                "description": "Specify deployment type: DevTest, POC, Prod, Shared. This will also be used in the naming convention."
+            }
+        },
+        "sqlAdministratorLogin": {
+            "type": "string",
+            "defaultValue": "sqladminuser",
+            "metadata": {
+                "description": "The username of the SQL Administrator"
+            }
+        },
+        "sqlAdministratorLoginPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "The password for the SQL Administrator"
+            }
+        },
+        "storageRoleUniqueId": {
+            "type": "string",
+            "defaultValue": "[newGuid()]",
```

Comment:
this must be idempotent and unique, and should be a variable (so the user can't break it)